### PR TITLE
core: improved consumption of audit.details

### PR DESCRIPTION
**Pull request by patrickhulce**
moves us toward #4614 (and #3115), really helps out using details data

@paulirish did you still want to tackle the rest or is this for me? :) either way should we bring the issue in to ocho and assign it?

### CODE REVIEW

File: lighthouse-core/report/v2/renderer/details-renderer.js [hunk 1]
```diff
@@ -59,6 +64,32 @@ class DetailsRenderer {
     }
   }
 
+  /**
+   * @param {!DetailsRenderer.NumericUnitDetailsJSON} details
+   * @return {!Element}
+   */
+  _renderBytes(details) {
+    let text = Util.formatNumber(details.value, details.granularity || 0);
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/report/v2/renderer/details-renderer.js]: this should never even happen, so I don't have a strong opinion on how it's handled but... creating a new helper function that is never called does seem a bit silly, so I'm inclined to say we just always do formatBytesToKB and console.warn if display unit isn't kb?

**Comment by patrickhulce** [comment]
> am kinda surprised we had no tests that asserted the details object. can you add some? alternatively we move those tests below to use details instead of extInfo.

that was because details is still fragile and annoying based on the fact that adding a new column would require rebasing all of our tests. IMO I still don't really see it as a wholesale replacement of extendedInfo in terms of ergonomics, if people rely on array indices to be stable they'll be more disappointed than just using extendedInfo

maybe we should work out exactly how we expect people to consume this data and what we want to guarantee as stable? doesn't seem like we can guarantee the text of the headers to be the same and certainly not the index of the column, so extendedInfo-style might actually be a better bet

**Comment by paulirish** [comment]
Remaining audits that need to be adapted: 

> 'link-blocking-first-paint',         'script-blocking-first-paint',         'time-to-first-byte',         'redirects',

File: lighthouse-core/audits/byte-efficiency/byte-efficiency-audit.js [hunk 2]
```diff
@@ -33,28 +32,27 @@ class UnusedBytes extends Audit {
...
-  /**
-   * @param {number} bytes
-   * @param {number} percent
-   * @return {string}
-   */
-  static toSavingsString(bytes = 0, percent = 0) {
-    const kbDisplay = this.bytesToKbString(bytes);
-    const percentDisplay = Util.formatNumber(Math.round(percent)) + '%';
-    return `${kbDisplay} (${percentDisplay})`;
+  static bytesToDetails(bytes) {
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/audits/byte-efficiency/byte-efficiency-audit.js]: done

**Comment by paulirish** [comment]
Here's my initial take at merging extInfo and details together and using proper key names: http://bit.ly/2HDIk8q (lol the branch name hit a github bug. :)

Also here's a LHR this branch produces: 
[tinyhouse.json.txt](https://github.com/GoogleChrome/lighthouse/files/1752641/tinyhouse.json.txt)

**Comment by paulirish** [comment]
This is LGTM. 
When we land, we'll have officially kicked off the 3.0 on `master`.

**Comment by paulirish** [comment]
### Hello 3️⃣ ⚫️ 0️⃣ !

**Pull request merged**